bin/runtime-common/src/messages_benchmarking.rs

```diff
@@ -291,3 +281,44 @@ where
 	pallet_bridge_parachains::initialize_for_benchmarks::<R, PI, PC>(bridged_header);
 	(bridged_block_number, bridged_header_hash)
 }
+
+/// Returns callback which generates `BridgeMessage` from polkadot xcm builder based on `expected_message_size`.
+pub fn generate_xcm_builder_bridge_message_sample(
+	destination: InteriorMultiLocation,
+) -> impl Fn(usize) -> Vec<u8> {
+	move |expected_message_size| -> Vec<u8> {
+		// For XCM bridge hubs, it is the message that
+		// will be pushed further to some XCM queue (XCMP/UMP)
+		let location = xcm::VersionedInteriorMultiLocation::V3(destination);
+		let location_encoded_size = location.encoded_size();
+
+		// we don't need to be super-precise with `expected_size` here
+		let xcm_size = expected_message_size.saturating_sub(location_encoded_size);
+		let xcm_number_of_instructions =
+			xcm_size.saturating_div(Instruction::<()>::ClearOrigin.encoded_size()) + 1;
+		log::trace!(
+			target: "runtime::bridge-benchmarks",
+			"generate_xcm_builder_bridge_message_sample with expected_message_size: {}, location_encoded_size: {}, xcm_size: {}, xcm_number_of_instructions: {}, ClearOrigin.encoded_size: {}",
+			expected_message_size, location_encoded_size, xcm_size, xcm_number_of_instructions, Instruction::<()>::ClearOrigin.encoded_size(),
+		);
+
+		// XCM V3 introduced limit for max instructions
+		const MAX_INSTRUCTIONS_TO_DECODE: u32 = 100;
+		let xcm_number_of_instructions =
+			sp_std::cmp::min(xcm_number_of_instructions, MAX_INSTRUCTIONS_TO_DECODE as usize);
+		log::trace!(
+			target: "runtime::bridge-benchmarks",
+			"generate_xcm_builder_bridge_message_sample using xcm_number_of_instructions: {}", xcm_number_of_instructions
+		);
+
+		let xcm = xcm::VersionedXcm::<()>::V3(
+			vec![Instruction::ClearOrigin; xcm_number_of_instructions].into(),
```

Comment:
Sorry - I'm too late to the party, but to generate more precise message, we could use the `ExpectPallet` pallet instruction instead of vec of `ClearOrigin`. Something like:
```rust
ExpectPallet {
		index: 0,
		name: vec![42; expected_message_size - <already-occupied-stuff>],
        module_name: vec![0],
       crate_major: 0,
       min_crate_minor: 0,
		module_name: Vec<u8>,
		#[codec(compact)]
		crate_major: u32,
		#[codec(compact)]
		min_crate_minor: u32,
	}
```

Reply:
no, no, you're not late, thats where I ended yesterday - to change ClearOrigin for some other instruction :D,
so thank you for `ExpectPallet` hint :)